purchase_lot/views/purchase_order_view.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+
+        <record id="purchase_order_lot_form" model="ir.ui.view">
+            <field name="model">purchase.order</field>
+            <field name="inherit_id" ref="purchase.purchase_order_form" />
+            <field name="arch" type="xml">
+                <xpath
+                expr="//field[@name='order_line']/tree/field[@name='name']"
+                position="after"
+            >
+                    <field name="lot_id" readonly="1" />
+                </xpath>
+            </field>
+        </record>
+
+        <record id="purchase_order_line_lot_form" model="ir.ui.view">
+            <field name="model">purchase.order.line</field>
+            <field name="inherit_id" ref="purchase.purchase_order_line_form2" />
+                <field name="arch" type="xml">
+                    <xpath expr="//field[@name='product_id']" position="after">
```

Comment:
We can replace the xpath expression by just using `<field name="product_id" position="after"></field>`